ui/perfherder/graphs/TestDataModal.jsx

```diff
@@ -465,6 +498,8 @@ export default class TestDataModal extends React.Component {
     let tests = [];
     if (filterText) {
       tests = filteredData;
+    } else if (tag !== this.defaultTag) {
```

Comment:
Why make a special case for filtering by tags? I see that `updateTaggedTests` & `updateFilterText` are coupled: the latter one expects the 1st one to run first (to properly fill in the `taggedTests` variable).

Would it be a good idea to include the logic from `updateTaggedTests` inside `updateFilterText`? (And maybe rename `updateFilterText` to something more appropriate? Hint: text to filter & tags are client side filters.)